### PR TITLE
Make an access to the client-side resources from server module

### LINKED ISSUE #4371: OGSI: BootstrapHandler being in flow-server wants to access to the resources in flow-client
`BootstrapHandler` class needs generated resource in the `flow-client` module.
The way how it's done is : 
```
BootstrapHandler.class.getResource(
                "/META-INF/resources/"+...);
```
Since the resources are in the different bundle it's not possible to access them in this way.
The code should be adapted to be able to work with OSGi.

The implementation suggestion:
- Make a service in `flow-server`.
- `flow-client` depends on the `flow-server` so it may implement a service and register it.
- `flow-server` should check the presence of OSGi and if we are in OGSi container use the service, otherwise use plain Java approach.
- we will need to have a code which depends on the OSGi classes which may not be in the classpath during runtime. So OSGi dependency should be added to the `flow-server`  module as `provided`  and OSGi related code should be separated from other code so that there will be no runtime exception on resources load.

That's one of the suggestion.
There can be other ways (don't know about them unfortunately) but I'm going to implement it in this way for now.
The implementation may be changed later on. But the issue is still the same: the resources are in the other bundle and we need a way to access them.

### PR DESCRIPTION
**Pull request by denis-anisimov**
Fixes #4371

### CODE REVIEW

**Comment by SomeoneToIgnore** [comment]
299 tests fail, guess it's better to have them fixed before doing a review.

**Comment by denis-anisimov** [comment]
Now everything should be fixed.

File: flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.client;
+
+import java.io.InputStream;
+import java.io.Serializable;
+
+/**
+ * Service which allows to get reosurces from the client side module.
```
Comment (SomeoneToIgnore) [inline flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java]: > reosurces

typo

File: flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+@FunctionalInterface
+public interface ClientResources extends Serializable {
+
+    /**
+     * Get content of the resource in the client-side module.
+     *
+     * @param path
+     *            the resource path
+     * @return the content of the resource as InutStream or null if there is no
```
Comment (SomeoneToIgnore) [inline flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java]: > InutStream

typo. Might worth to make this as a `{@link` and null as `{@code`, but up to you.

File: flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java (hunk 1 above)
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java]: Done

File: flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java (hunk 2 above)
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java]: typo is fixed, `null` is fixed.
`link`  does not work in javadoc inside `@param`/`@return`

Comment (SomeoneToIgnore) [reply flow-server/src/main/java/com/vaadin/flow/client/ClientResources.java]: Oh, did not notice it, good.

**Comment by SomeoneToIgnore** [comment]
One file conflicts with `master` now, otherwise good.

**Pull request merged**